### PR TITLE
LibJS: A couple of Temporal editorial simplifications

### PR DESCRIPTION
**Pull request by IdanHo**
Ticks 2 boxes off #15525.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 1]
```diff
@@ -1666,41 +1666,30 @@ ThrowCompletionOr<TemporalTime> parse_temporal_time_string(VM& vm, String const&
...
+// 13.38 ParseTemporalTimeZoneString ( timeZoneString ), https://tc39.es/proposal-temporal/#sec-temporal-parsetemporaltimezonestring
+ThrowCompletionOr<TemporalTimeZone> parse_temporal_time_zone_string(VM& vm, String const& time_zone_string)
 {
-    // 1. Let parseResult be ParseText(StringToCodePoints(isoString), TemporalTimeZoneString).
-    auto parse_result = parse_iso8601(Production::TemporalTimeZoneString, iso_string);
-
-    // 2. If parseResult is a List of errors, throw a RangeError exception.
-    if (!parse_result.has_value())
-        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidTimeZoneString, iso_string);
+    // 1. Let parseResult be ParseText(StringToCodePoints(timeZoneString), TemporalTimeZoneString).
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1672]:
```suggestion
    // 1. Let parseResult be ParseText(StringToCodePoints(timeZoneString), TimeZoneIdentifier).
```

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 2]
```diff
@@ -1666,41 +1666,30 @@ ThrowCompletionOr<TemporalTime> parse_temporal_time_string(VM& vm, String const&
     return TemporalTime { .hour = result.hour, .minute = result.minute, .second = result.second, .millisecond = result.millisecond, .microsecond = result.microsecond, .nanosecond = result.nanosecond, .calendar = move(result.calendar) };
 }
 
-// 13.38 ParseTemporalTimeZoneString ( isoString ), https://tc39.es/proposal-temporal/#sec-temporal-parsetemporaltimezonestring
-ThrowCompletionOr<TemporalTimeZone> parse_temporal_time_zone_string(VM& vm, String const& iso_string)
+// 13.38 ParseTemporalTimeZoneString ( timeZoneString ), https://tc39.es/proposal-temporal/#sec-temporal-parsetemporaltimezonestring
+ThrowCompletionOr<TemporalTimeZone> parse_temporal_time_zone_string(VM& vm, String const& time_zone_string)
 {
-    // 1. Let parseResult be ParseText(StringToCodePoints(isoString), TemporalTimeZoneString).
-    auto parse_result = parse_iso8601(Production::TemporalTimeZoneString, iso_string);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1673]: I think `TemporalTimeZoneString` is unused after this, let's remove it from the public parsing interface

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1673]: Actually, it seems to be gone completely from the spec, I wonder where they removed it...

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1672]: whoops! fixed.

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp (hunk 2 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1673]: Sure, would you prefer a separate commit?

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1673]: Aha, it's removed in https://github.com/tc39/proposal-temporal/commit/eec8efab, so just add that to the second commit :)

Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1673]: done!

**Review by linusg — APPROVED** [review]
Nice!

**Pull request merged**